parsl/dataflow/dflow.py

```diff
@@ -496,17 +497,25 @@ def _add_input_deps(self, executor, args, kwargs):
 
         inputs = kwargs.get('inputs', [])
         for idx, f in enumerate(inputs):
+            if isinstance(f, DataFuture) and f.file_obj.is_remote():
+                inputs[idx] = self.data_manager.stage_in(f.file_obj, executor, f)
             if isinstance(f, File) and f.is_remote():
-                inputs[idx] = self.data_manager.stage_in(f, executor)
+                inputs[idx] = self.data_manager.stage_in(f, executor, None)
+            # if neither of the above, then local File, or local DataFuture, or non-File-related
+            # so do no stage-in
 
         for kwarg, f in kwargs.items():
-            if isinstance(f, File) and f.is_remote():
-                kwargs[kwarg] = self.data_manager.stage_in(f, executor)
+            if isinstance(f, DataFuture) and f.file_obj.is_remote():
```

Comment:
Consider making `parent_fut` required and performing the `isinstance(f, DataFuture)`/`isinstance(f, File)` check inside `data_manager.stage_in`-- then we avoid the repeated checks below